### ISSUE TITLE
Cross-layer predictor: empirically the unlock for large-MoE regime (235B test)

### ISSUE THREAD

**Issue by vvsotnikov** [vvsotnikov/llama.cpp#3]
## Problem
Temporal-1 predictor ("use last step's experts") gives ~78% recall. Step-1's offline analysis showed a much better predictor exists:
- **Cross-layer**: layer M's gate applied to layer M-1's residual → **88% k=1 recall**.
- With **top-16 overfetch**: **98.5% recall** (approaches oracle's 97.7%).

## Why it matters
Going from 78% → 98.5% recall reduces the "substitute-and-go" misses that corrupt output. Current C=32 live predictor degrades after ~12 tokens; with 98% recall it would stay coherent indefinitely, matching oracle quality.

## Implementation sketch
Per-layer in `build_moe_ffn`:
```cpp
// After layer M's `ffn_moe_logits` is computed:
predicted_logits_M_plus_1 = build_lora_mm(ffn_gate_inp[M+1], hidden_M);
predicted_topk = ggml_argsort_top_k(predicted_logits_M_plus_1, K + overfetch);
```
The predicted_topk for layer M+1 is then observed via cb_eval (or via the better mechanism from #2) and fed to `predictor_prefill` for the NEXT decode.

Step-1 reference: `~/moe-prefetch-poc/analyze.py` has the offline accuracy numbers.

## Acceptance
- Predictor recall (measured against oracle on same prompts) ≥ 85% k=1.
- With top-16 overfetch ≥ 95% effective hit at C=32.

Depends on #2 (faster observation) to be useful at runtime.
